### PR TITLE
Modal component #8

### PR DESCRIPTION
**Pull request by Shinji22**
Modal component, including 3 sub-components (Modal-header, Modal-body and Modal-footer)

### CODE REVIEW

File: src/packages/Modal/styles.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import styled, { css } from 'styled-components'
+import { media } from '../../styles'
+import { ModalFooterProps, ModalProps } from '.'
+
+const modalVariants = {
+  small: () => css`
+    width: 100%;
+    ${media.xs`
+			width: 300px;
```
Comment (vczb) [inline src/packages/Modal/styles.ts:9]: 30rem

File: src/packages/Modal/styles.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    `}
+  `,
+  medium: () => css`
+    width: 100%;
+    ${media.xs`
+			width: 300px;
```
Comment (vczb) [inline src/packages/Modal/styles.ts:15]: 30rem

File: src/packages/Modal/styles.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    `}
+    ${media.sm`
+			width: 500px;
```
Comment (vczb) [inline src/packages/Modal/styles.ts:18]: 50rem

File: src/packages/Modal/styles.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    `}
+  `,
+  large: () => css`
+    width: 100%;
+    ${media.xs`
+			width: 300px;
```
Comment (vczb) [inline src/packages/Modal/styles.ts:24]: 30rem

File: src/packages/Modal/styles.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    `}
+    ${media.sm`
+			width: 500px;
```
Comment (vczb) [inline src/packages/Modal/styles.ts:27]: 50rem

File: src/packages/Modal/styles.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    `}
+    ${media.md`
+			width: 800px;
```
Comment (vczb) [inline src/packages/Modal/styles.ts:30]: 80rem

**Review by vczb — COMMENTED** [review]
Hi @Shinji22 I'm very happy to see that you use correctly the theme and the media queries. I already check your component and it is very beautiful and useful.

The only request-change needed here is update the `pixels` to `rem` , I left some comments on the places that need to change. 

---

everywhere you used pixel you can divide the value by 10, for example: 300 becomes 30

**Comment by vczb** [comment]
Also, you can squash all commits in a single one ?

image


If you have any question I'm available to help

File: src/packages/Modal/styles.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+      box-sizing: border-box;
+      ${size ? modalVariants[size]() : modalVariants.medium()}
+    }
+  `}
+`
+
+export const ModalBody = styled.div`
+  ${({ theme }) => css`
+    flex-grow: 1;
+    min-height: 100px;
```
Comment (vczb) [inline src/packages/Modal/styles.ts:67]: 10rem

**Review by vczb — APPROVED** [review]
LGTM! :1st_place_medal: 

Please squash all commits in a single one

**Comment by Shinji22** [comment]
All commits are squash now

**Pull request merged**